Update go version to 1.19

### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to lotus.

### Lotus component

- [X] lotus daemon - chain sync
- [ ] lotus fvm/fevm - Lotus FVM and FEVM interactions
- [X] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt/WinningPoSt)
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### Lotus Version

```text
v1.21.0-rc1
```


### Repro Steps

Try to build the docker image of lotus.

```
docker build --build-arg LOTUS_TEST_IMAGE=filecoin/lotus-all-in-one:v1.21.0-rc1-debug --build-arg FFI_BUILD_FROM_SOURCE=0  -f Dockerfile --target lotus-all-in-one \
	-t filecoin/lotus-all-in-one:v1.21.0-rc1-debug --build-arg GOFLAGS=-tags=debug .
```

### Describe the Bug

When I try to build a lotus docker image, it fails with below error:

```
Step 12/79 : RUN make clean deps
 ---> Running in df52d903b86e
Makefile:14: Your Golang version is go1.18.8
Makefile:15: *** Update Golang to version to at least 1.19.7.  Stop.
The command '/bin/sh -c make clean deps' returned a non-zero code: 2
make: *** [Makefile:232: docker/lotus-all-in-one] Error 2
```

I think requirement is coming from one of the dependencies and as long as user builds Lotus manually outside docker, it will work if they are using go v1.19.x
But, docker is strict in the sense it will use defined image of v1.18.x in the Dockerfile.

### Logging Information

```text
No required
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,5 +1,5 @@
 #####################################
-FROM golang:1.18.8-buster AS lotus-builder
+FROM golang:1.19.7-buster AS lotus-builder
 MAINTAINER Lotus Development Team
 
 RUN apt-get update && apt-get install -y ca-certificates build-essential clang ocl-icd-opencl-dev ocl-icd-libopencl1 jq libhwloc-dev

--- a/Dockerfile.lotus
+++ b/Dockerfile.lotus
@@ -1,6 +1,6 @@
 ##### DEPRECATED
 
-FROM golang:1.18.8-buster AS builder-deps
+FROM golang:1.19.7-buster AS builder-deps
 MAINTAINER Lotus Development Team
 
 RUN apt-get update && apt-get install -y ca-certificates build-essential clang ocl-icd-opencl-dev ocl-icd-libopencl1 jq libhwloc-dev

--- a/README.md
+++ b/README.md
@@ -71,10 +71,10 @@ For other distributions you can find the required dependencies [here.](https://l
 
 #### Go
 
-To build Lotus, you need a working installation of [Go 1.18.8 or higher](https://golang.org/dl/):
+To build Lotus, you need a working installation of [Go 1.19.7 or higher](https://golang.org/dl/):
 
 ```bash
-wget -c https://golang.org/dl/go1.18.8.linux-amd64.tar.gz -O - | sudo tar -xz -C /usr/local
+wget -c https://golang.org/dl/go1.19.7.linux-amd64.tar.gz -O - | sudo tar -xz -C /usr/local
 ```
 
 **TIP:**

